mono/metadata/locales.c

```diff
@@ -559,7 +559,8 @@ get_current_locale_name (void)
 #ifdef HOST_WIN32
 	locale = g_win32_getlocale ();
 #elif defined (__APPLE__)	
-	locale = get_darwin_locale ();
+	if (!g_hasenv ("MONO_DARWIN_FAST_LOCALE"))
```

Comment:
I'd name this `MONO_DARWIN_PREFER_POSIX_LOCALE`.

We'll also need to document it in the man page.

Reply:
The POSIX locale is based on environment variables and might not be set when launching outside of a shell (launchd, the open command, iOS apps).